src/python/pants/backend/go/target_types.py

```diff
@@ -226,19 +224,6 @@ class GoBinaryMainPackageField(StringField, AsyncFieldMixin):
     value: str
 
 
-@dataclass(frozen=True)
-class GoBinaryMainPackage:
-    address: Address
-
-
-@dataclass(frozen=True)
-class GoBinaryMainPackageRequest(EngineAwareParameter):
-    field: GoBinaryMainPackageField
-
-    def debug_hint(self) -> str:
-        return self.field.address.spec
-
-
```

Comment:
These have been moved to `util_rules/binary.py` to avoid a circular import due to `util_rules/go_mod.py` now using these.